src/config.rs

```diff
@@ -0,0 +1,54 @@
+use crate::db::DBVersion;
+use crate::dir::Rank;
+
+use anyhow::{bail, Context, Result};
+
+use std::env;
+use std::fs;
+use std::path::PathBuf;
+
+pub const DB_MAX_SIZE: u64 = 8 * 1024 * 1024; // 8 MiB
+pub const DB_VERSION: DBVersion = 3;
+
+pub fn zo_data_dir() -> Result<PathBuf> {
+    let data_dir = match env::var_os("_ZO_DATA_DIR") {
+        Some(data_osstr) => PathBuf::from(data_osstr),
+        None => {
+            if let Some(mut data_dir) = dirs::data_local_dir() {
+                data_dir.push("zoxide");
+                data_dir
+            } else {
+                bail!("could not find database directory, please set _ZO_DATA_DIR manually");
+            }
+        }
```

Comment:
You asked me to `match` on something similar to this in my rework of the fallback. Maybe we should do the same here (feel free to rebase it into the rework commit).

```suggestion
        None => match dirs::data_local_dir() {
            Some(mut data_dir) => {
                data_dir.push("zoxide");
                data_dir
            None => bail!("could not find database directory, please set _ZO_DATA_DIR manually"),
        }
```